Packs/PcapAnalysis/Scripts/PcapFileExtractor/PcapFileExtractor.py

```diff
@@ -0,0 +1,195 @@
+    stdout = str(stdout).split('\\n')
+    for line in stdout:
+        if line.endswith(')') and 'HTTP' in line:
+            data_list.append(line)
+            protocol = 'http'
+
+        if line.endswith(')') and 'IMF' in line:
+            data_list.append(line)
+            protocol = 'imf'
+
+        if line.endswith(')') and 'SMB' in line:
+            data_list.append(line)
+            protocol = 'smb'
+
+        if line.endswith(')') and 'DICOM' in line:
+            data_list.append(line)
+            protocol = 'dicom'
+
+        if line.endswith(')') and 'TFTP' in line:
+            data_list.append(line)
+            protocol = 'tftp'
+
+    return protocol, data_list
+
+
+def extract_files(file_path, dir_path):
+    """
+    :param file_path: the pcap file path.
+    :param dir_path: directory for the files will be extract to
+    :return:
+        excluded_files: the excludedfiles which are in dir_path
+    """
+    protocol, _ = find_files_protocol(file_path)
+    if not protocol:
+        return_error('Could not find a valid protocol for extracting the files')
+
+    if protocol == 'http':
+        command_ = f'tshark -r {file_path} --export-objects http,{dir_path}'
+
+    elif protocol == 'smb':
+        command_ = f'tshark -r {file_path} --export-objects smb,{dir_path}'
+
+    elif protocol == 'imf':
+        command_ = f'tshark -r {file_path} --export-objects imf,{dir_path}'
+
+    elif protocol == 'tftp':
+        command_ = f'tshark -r {file_path} --export-objects tftp,{dir_path}'
+
+    elif protocol == 'dicom':
+        command_ = f'tshark -r {file_path} --export-objects dicom,{dir_path}'
+
+    process = subprocess.Popen(command_, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    process.communicate()
+    excluded_files = [f for f in glob.glob(dir_path + '/*')]
+
+    return excluded_files
+
+
+def upload_files(excluded_files, dir_path, file_path):
+    """
+
+    :param excluded_files: excluded files
+    :param dir_path: dir path for the files
+    :param file_path: the path to the pcap file
+    :return:
+    """
+    filenames = []  # type: ignore
+    # recursive call over the file system top down
+    for root, directories, files in os.walk(dir_path):
+        for f in files:
+            # skipping previously existing files
+            # adding it to the extracted pcap files list
+            if f not in excluded_files and isfile(os.path.join(root, f)):
+                filenames.append(os.path.join(root, f))
+
+    if len(filenames) == 0:
+        return_error('Could not find files')
+
+    else:
+        results = []
+        context = []
+        protocol, packet_data = find_files_protocol(file_path)
+
+        md5 = hashlib.md5()
+        sha1 = hashlib.sha1()
+        sha256 = hashlib.sha256()
+        files_base_names = [os.path.basename(file_path) for file_path in filenames]  # noqa[F812]
+        files_dic = {file_path: os.path.basename(file_path) for file_path in filenames}
```

Comment:
not sure why we're iterating the same object 3 times, twice here and 1 in the loop below
lets just iterate it once